src/blockdev.rs

```diff
@@ -136,7 +136,7 @@ impl Disk {
                 .write(true)
                 .open(&self.path)
                 .with_context(|| format!("opening {}", &self.path))?;
-            reread_partition_table(&mut f).map(|_| Vec::new())
+            reread_partition_table(&mut f, true).map(|_| Vec::new())
```

Comment:
I think we shouldn't be retrying here.  We're not calling this to update anything after a change; we're calling it exactly because we want to know whether it fails on a busy partition.